examples/module/chardev/chardev.c

```diff
@@ -101,12 +101,13 @@ static ssize_t chardev_write(FAR struct file *filep, FAR const char *buffer,
  * Name: module_uninitialize
  ****************************************************************************/
 
-static int module_uninitialize(FAR void *arg)
+__attribute__((destructor))
```

Comment:
> Isn't this specific to GCC compiler.  One of the basic rules is that we never include compiler-specific directories in common code that may use a different toolchain.

The correct way to handle this is to put a defininition in include/nuttx/compiler.h that defines a toolchain agnostic symbol for this.  For toolchains that do not support that attribute, that symbol can be defined as nothing.

Refer to "All Users Matter" in the to-level INVIOLABLES.txt